imports/plugins/core/inventory/server/no-meteor/utils/publishProductToCatalog.js

```diff
@@ -9,20 +9,47 @@ export default async function publishProductToCatalog(catalogProduct, { context,
 
-  const topVariantsInventoryInfo = await context.queries.inventoryForProductConfigurations(context, {
+  // Retrieve inventory information for all top level variants
+  const topVariantsAndOptionsInventory = await context.queries.inventoryForProductConfigurations(context, {
     fields: ["isBackorder", "isLowQuantity", "isSoldOut"],
-    productConfigurations: topVariants.map((option) => ({
+    productConfigurations: topVariantsAndTopOptions.map((option) => ({
       isSellable: !variants.some((variant) => variant.ancestors.includes(option._id)),
       productId: option.ancestors[0],
       productVariantId: option._id
     })),
     shopId: catalogProduct.shopId
   });
 
-  // Mutate the catalog product to be saved
-  catalogProduct.isBackorder = topVariantsInventoryInfo.every(({ inventoryInfo }) => inventoryInfo.isBackorder);
-  catalogProduct.isLowQuantity = topVariantsInventoryInfo.some(({ inventoryInfo }) => inventoryInfo.isLowQuantity);
-  catalogProduct.isSoldOut = topVariantsInventoryInfo.every(({ inventoryInfo }) => inventoryInfo.isSoldOut);
+  // Add inventory properties to the top level parent product.
+  catalogProduct.isBackorder = topVariantsAndOptionsInventory.every(({ inventoryInfo }) => inventoryInfo.isBackorder);
+  catalogProduct.isLowQuantity = topVariantsAndOptionsInventory.some(({ inventoryInfo }) => inventoryInfo.isLowQuantity);
+  catalogProduct.isSoldOut = topVariantsAndOptionsInventory.every(({ inventoryInfo }) => inventoryInfo.isSoldOut);
+
+  // add inventory props to each top level Variant
+  catalogProduct.variants.forEach((variant, variantIndex) => {
+    // attempt to find this variant's inventory info
+    const foundVariantInventory = topVariantsAndOptionsInventory.find((inventoryInfo) => inventoryInfo.productConfiguration.productVariantId === variant._id);
+
+    // if inventory info was found, add to variant
+    if (foundVariantInventory) {
+      catalogProduct.variants[variantIndex].isBackorder = foundVariantInventory.inventoryInfo.isBackorder;
+      catalogProduct.variants[variantIndex].isLowQuantity = foundVariantInventory.inventoryInfo.isLowQuantity;
+      catalogProduct.variants[variantIndex].isSoldOut = foundVariantInventory.inventoryInfo.isSoldOut;
+    }
+
+    // add inventory props to each top level option
+    variant.options.forEach((option, optionIndex) => {
+      const foundOptionInventory = topVariantsAndOptionsInventory.find((inventoryInfo) => inventoryInfo.productConfiguration.productVariantId === option._id);
+
+      // if inventory info was found, add to variant
+      if (foundOptionInventory) {
+        catalogProduct.variants[variantIndex].options[optionIndex].isBackorder = foundOptionInventory.inventoryInfo.isBackorder;
+        catalogProduct.variants[variantIndex].options[optionIndex].isLowQuantity = foundOptionInventory.inventoryInfo.isLowQuantity;
+        catalogProduct.variants[variantIndex].options[optionIndex].isSoldOut = foundOptionInventory.inventoryInfo.isSoldOut;
```

Comment:
I believe `option` is by reference, so you should be able to do `option` instead of `catalogProduct.variants[variantIndex].options[optionIndex]` here.